Don't change interfaces unless there is something wrong with the old one.
This is NTP bug [2043] and this patch should be removed after upgrading to
a version of NTP with a fix for that bug.

--- ntpd/ntp_peer.c
+++ ntpd/ntp_peer.c
@@ -718,9 +718,13 @@
 	/*
 	 * this is called when the interface list has changed
 	 * give all peers a chance to find a better interface
+	 * but only if either they don't have an address already
+	 * or if the one they have hasn't worked for a while.
 	 */
-	for (p = peer_list; p != NULL; p = p->p_link)
-		peer_refresh_interface(p);
+	for (p = peer_list; p != NULL; p = p->p_link) {
+		if (!(p->dstadr && (p->reach & 0x3)))
+			peer_refresh_interface(p);
+	}
 }
 
 
